serverless-workflow-examples/serverless-workflow-compensation-quarkus/src/main/resources/compensation.sw.json

```
@@ -4,64 +4,95 @@
   "name": "Workflow Error example",
   "description": "An example of how compensation works",
   "start": "printStatus",
+  "functions": [
+    {
+      "name": "PrintOutput",
+      "type": "custom",
+      "operation": "sysout"
+    }
+  ],
   "states": [
- 	{
+    {
       "name": "printStatus",
       "type": "inject",
       "data": {
-         "compensated": false
+        "compensated": false
       },
-      "compensatedBy" : "compensating",
-       "transition": "branch"
-    }, 
+      "compensatedBy": "compensating",
+      "transition": "branch"
+    },
     {
       "name": "branch",
       "type": "switch",
       "dataConditions": [
         {
           "condition": ".shouldCompensate==true",
-          "transition": { 
-            "nextState" : "finish_compensate",
-            "compensate" : true
-           }
-        }, 
+          "transition": {
+            "nextState": "finish_compensate",
+            "compensate": true
+          }
+        },
         {
           "condition": ".shouldCompensate==false",
-          "transition": { 
-            "nextState" : "finish_not_compensate",
-            "compensate" : false
-           }
+          "transition": {
+            "nextState": "finish_not_compensate",
+            "compensate": false
+          }
         }
-       ]
+      ],
+      "defaultCondition": {
+        "end": true
+      }
     },
     {
       "name": "compensating",
-      "usedForCompensation" : true,
+      "usedForCompensation": true,
       "type": "inject",
       "data": {
         "compensated": true
-      }, 
-      "transition" : "compensating_more"
+      },
+      "transition": "compensating_more"
     },
     {
       "name": "compensating_more",
-      "usedForCompensation" : true,
+      "usedForCompensation": true,
       "type": "inject",
       "data": {
         "compensating_more": "Real Betis Balompie"
-      }
+      },
+      "end": true
     },
     {
       "name": "finish_compensate",
       "type": "operation",
-      "actions": [],
+      "actions": [
+        {
+          "name": "finish_compensate_sysout",
+          "functionRef": {
+            "refName": "PrintOutput",
+            "arguments": {
+              "message": "completed"
```

```suggestion
              "message": "Compensate completed"
```
to distinguish `finish_compensate` and `finish_not_compensate`?

---

can be.

---

I think we should not add the print action at all and leave the flow as it was (fixing the fomatting if we want but thats it)

---

In fact, besides the readme change (removing the quotes for the boolean, which was a typo), I do not think this PR should have been opened

---

if you think that the action should not do nothing, it is even better to remove that, at least for me, have empty action list does not make sense.

imho what we can do is, remove the action, change to something else or keep the action with sysout.

What we can't have is one sdk accepting empty array list while the spec mentions that action can't be null, while the other sdk, requires at least one action.

---

To satisfy the spec and to showcase the compensation, I think this is OK.